Cabal/src/Distribution/Lex.hs

```diff
@@ -16,6 +16,14 @@ import Distribution.Compat.DList
 import Distribution.Compat.Prelude
 import Prelude ()
 
+-- | A simple parser supporting quoted strings.
+--
+-- Please be aware that this will only split strings when seeing whitespace
+-- outside of quotation marks, i.e, "foo\"bar baz\"qux quux" will be
+-- converted to ["foobar bazqux", "quux"].
+--
+-- This behavior can be useful when parsing text like "ghc-options=\"..\"",
```

Comment:
It's `ghc-options:` followed by a list of tokens which may be quoted shell-fashion, not `ghc-options=...`.